src/screens/CategoryDetailScreen/CategoryDetailScreen.tsx

```diff
@@ -0,0 +1,122 @@
+import React, {useContext} from 'react';
+import {FlatList, ListRenderItemInfo, StyleSheet} from 'react-native';
+import {CategoryInterface, RootStoreContext} from '../../models';
+import {CategoryDetailScreenProps} from './CategoryDetailScreen.types';
+import {CategoryItem} from './components';
+import {
+  DeviceUtils,
+  MoreCategories,
+  horizontalScale,
+  verticalScale,
+} from '../../utils';
+import {useNavigation} from '@react-navigation/native';
+import {NativeStackNavigationProp} from '@react-navigation/native-stack';
+import {PrimaryStackParams, PrimaryStackRoute} from '../../navigation';
+import {useThemeColor} from '../../hooks';
+import {fontStyles} from '../../constants';
+
+const CategoryDetailScreen = (props: CategoryDetailScreenProps) => {
+  const {alias} = props.route.params;
+  const {categories} = useContext(RootStoreContext);
+  const {colors} = useThemeColor();
+
+  const navigation =
+    useNavigation<NativeStackNavigationProp<PrimaryStackParams>>();
+
+  const checkSubCategories = (categoryAlias: string) => {
+    return (
+      categories.allCategories.filter(category =>
+        category.parent_aliases.includes(categoryAlias),
+      ).length !== 0
+    );
+  };
+
+  const renderItem = (
+    renderItemProps: ListRenderItemInfo<CategoryInterface>,
+    showIcon: boolean,
+  ) => {
+    const {item, index} = renderItemProps;
+
+    const onPressHandler = () => {
```

Comment:
Avoid functions within functions

Reply:
Replaced with useCallbacks by seeing this answer https://stackoverflow.com/a/69732518